cpp/dolfinx/fem/assemble_vector_impl.h

```diff
@@ -480,27 +517,33 @@ void _lift_bc_interior_facets(
 /// positive the block size is used as a compile-time constant, which
 /// has performance benefits.
 template <dolfinx::scalar T, int _bs = -1>
-void assemble_cells(fem::DofTransformKernel<T> auto dof_transform,
-                    std::span<T> b, mdspan2_t x_dofmap,
-                    std::span<const scalar_value_type_t<T>> x,
-                    std::span<const std::int32_t> cells, mdspan2_t dofmap,
-                    int bs, FEkernel<T> auto kernel,
-                    std::span<const T> constants, std::span<const T> coeffs,
-                    int cstride, std::span<const std::uint32_t> cell_info)
+void assemble_cells(
+    fem::DofTransformKernel<T> auto dof_transform, std::span<T> b,
+    mdspan2_t x_dofmap, std::span<const scalar_value_type_t<T>> x,
+    std::span<const std::int32_t> cells,
+    std::tuple<mdspan2_t, int, std::span<const std::int32_t>> dofmap,
```

Comment:
True. Since matrix assembly has already been merged using the same approach, maybe we could make an issue and address the best consistent approach in a future PR?